app/bin/service/search.dart

```diff
@@ -29,27 +29,19 @@ import 'package:pub_dartlang_org/search/updater.dart';
 final Logger _logger = new Logger('pub.search');
 
 Future main() async {
-  final ReceivePort errorPort = new ReceivePort();
-  errorPort.listen((error) {
-    print('ERROR from isolate: $error');
-  });
-  for (int i = 0; i < envConfig.isolateCount; i++) {
-    await Isolate.spawn(
-      _mainWrapper,
-      i,
-      onError: errorPort.sendPort,
-      onExit: errorPort.sendPort,
-      errorsAreFatal: true,
-    );
-  }
+  await startIsolates(_logger, _mainWrapper);
 }
 
 // Remove after https://github.com/dart-lang/sdk/issues/30755 lands.
 void _mainWrapper(message) => _main(message);
 
-void _main(int isolateId) {
+void _main(List<SendPort> sendPorts) {
   useLoggingPackageAdaptor();
 
+  final SendPort mainSendPort = sendPorts[0];
+  final ReceivePort taskReceivePort = new ReceivePort();
```

Comment:
We're creating another `taskReceivePort` in the code below. Should it use this port instead?

Reply:
good catch, fixed. (although in search case it is not used at the moment)